### PR TITLE
saul: add a self test function in saul driver

### PR DESCRIPTION
**Pull request by vincent-d**
Hi,

I would like to have your feedback on this.
I'm currently working on implementing some hardware tests on our board and having such a functionality for the saul interface could be interesting.
This PR just adds a test function to the saul interface which can be used to test the device. It's implemented on the mma8652 driver.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
Good idea, though I'd like to wait for @haukepetersen's feedback as he is the "father" of `saul`. Unfortunately he won't be available for the next couple of weeks

File: drivers/isl29020/isl29020_saul.c [hunk 1]
```diff
@@ -41,5 +41,6 @@ static int write(void *dev, phydat_t *state)
 const saul_driver_t isl29020_saul_driver = {
     .read = read,
     .write = write,
+    .test = saul_test_notsup,
```
Comment (jfischer-phytec-iot) [inline drivers/isl29020/isl29020_saul.c]: s/saul_test_notsup/NULL ?

File: sys/saul_reg/saul_reg.c [hunk 2]
```diff
@@ -126,3 +128,12 @@ int saul_reg_write(saul_reg_t *dev, phydat_t *data)
...
     return dev->driver->write(dev->dev, data);
 }
+
+int saul_reg_test(saul_reg_t *dev)
+{
+
+    if (dev == NULL) {
+        return -ENODEV;
+    }
+    return dev->driver->test(dev->dev);
```
Comment (jfischer-phytec-iot) [inline sys/saul_reg/saul_reg.c]: If you add a check for the NULL pointer `` if (dev->driver->test != NULL)`` here, then you do not need  the dummy function ``int saul_notsup(void *dev, phydat_t *dat)``

Comment (vincent-d) [inline sys/saul_reg/saul_reg.c]: I just did it the same way than the other calls. I can change if you prefer a NULL pointer instead of the stub.

**Comment by haukepetersen** [comment]
Actually, I  don't think this is needed in the SAUL interface. 

I see it the following: The device test should be integral part of a devices init function. So during init, it should be checked, if a device is connected correctly and any internal tests that could be done. All the `_test()` functions we have currently in RIOT do only the connection setup check, and this should be done during `init()`. 

Also it is quite important, that we keep the SAUL interface as slim as possible, as every added function leads to used memory through the needed pointer(s)....

**Comment by vincent-d** [comment]
Let's close this.

**Pull request closed without merging**